ptf/saiacl.py

```diff
@@ -0,0 +1,5852 @@
+                                     ip_dst=ip_addr,
+                                     ip_src=ip_addr_src,
+                                     tcp_sport=l4_dst_port,
+                                     ip_id=105,
+                                     ip_ttl=64)
+        self.exp_pkt = simple_tcp_packet(eth_dst=dmac,
+                                         eth_src=ROUTER_MAC,
+                                         ip_dst=ip_addr,
+                                         ip_src=ip_addr_src,
+                                         tcp_sport=l4_dst_port,
+                                         ip_id=105,
+                                         ip_ttl=63)
+
+    def runTest(self):
+        print("Testing DstIpAclTest")
+
+        print('--------------------------------------------------------------'
+              '--------------------------------')
+        print("Sending packet ptf_intf 2 -> ptf_intf 1 (192.168.100.100 "
+              "---> 172.16.10.1 [id = 105])")
+
+        try:
+            print('#### NO ACL Applied ####')
+            print('#### Sending  ', ROUTER_MAC, '| 00:22:22:22:22:22 | 172.16'
+                  '.10.1 | 192.168.100.100 | SPORT 1000 | @ ptf_intf 2')
+            send_packet(self, self.dev_port11, self.pkt)
+            print('#### Expecting 00:11:22:33:44:55 |', ROUTER_MAC, '| 172.16'
+                  '.10.1 | 192.168.100.100 | SPORT 1000 | @ ptf_intf 1')
+            verify_packets(self, self.exp_pkt, [self.dev_port10])
+        finally:
+            print('----------------------------------------------------------')
+
+        print("Sending packet ptf_intf 2 -[acl]-> ptf_intf 1 (192.168.0.1"
+              "-[acl]-> 172.16.10.1 [id = 105])")
+
+        # setup ACL to block based on Source IP and SPORT
+        table_stage_ingress = SAI_ACL_STAGE_INGRESS
+        table_stage_egress = SAI_ACL_STAGE_EGRESS
+        entry_priority = 1
+        ip_dst = "172.16.10.1"
+        ip_dst_mask = "255.255.255.0"
+
+        table_bind_points = [SAI_ACL_BIND_POINT_TYPE_ROUTER_INTF]
+        table_bind_point_type_list = sai_thrift_s32_list_t(
+            count=len(table_bind_points), int32list=table_bind_points)
+
+        acl_ingress_table_id = sai_thrift_create_acl_table(
+            self.client,
+            acl_stage=table_stage_ingress,
+            acl_bind_point_type_list=table_bind_point_type_list,
+            field_src_ip=True)
+
+        self.assertNotEqual(acl_ingress_table_id, 0)
+
+        dst_ip_t = sai_thrift_acl_field_data_t(
+            data=sai_thrift_acl_field_data_data_t(ip4=ip_dst),
+            mask=sai_thrift_acl_field_data_mask_t(ip4=ip_dst_mask))
+
+        packet_action = sai_thrift_acl_action_data_t(
+            parameter=sai_thrift_acl_action_parameter_t(
+                s32=SAI_PACKET_ACTION_DROP))
+
+        u32range = sai_thrift_u32_range_t(min=1000, max=1000)
+        acl_range_id = sai_thrift_create_acl_range(
+            self.client,
```

Comment:
Please update the comment and test name as this includes both L4 port range and dest-ip. Update other tests as well

Reply:
L3AclRangeTest I have separate test case for ranges. Maybe I should delete ranges form cases like src, dst, vlan or counter tests? I rewrote it from the old framework, it can be redundant. What do you think?